sklearn/tree/_splitter.pyx

```diff
@@ -519,16 +516,6 @@ cdef class BestSplitter(BaseDenseSplitter):
             for p in range(start, end):
                 sample_mask[samples[p]] = 0
 
-        # Respect invariant for constant features: the original order of
-        # element in features[:n_known_constants] must be preserved for sibling
-        # and child nodes
-        memcpy(features, constant_features, sizeof(SIZE_t) * n_known_constants)
-
```

Comment:
I'm not familiar with this code base at all, but it seems that `constant_features` is used here.

But if `constant_features` is never initialized maybe there is bug here.